### PR TITLE
test: adversarial unit-test sweep with latent bug fixes

### PR DESCRIPTION
**Pull request by yjsoon**
## Summary

Adversarial unit-test sweep: ~1,700 lines of new tests across 14 suites (edge cases, malformed inputs, races), plus fixes for latent production bugs the sweep exposed:

- **FrameStore** — a corrupt or unreadable `manifest.json` is quarantined (together with the frames directory, as `manifest.json.corrupt` / `frames.corrupt`) instead of bricking capture on every launch; the quarantine is logged to the diagnostics log and pre-corruption JPEGs are permanently out of reach of orphan cleanup
- **TextCache** — SQLite busy timeout so a connection opened during another connection's close-checkpoint doesn't fail init and silently disable OCR search for the session (verified with a standalone repro: 198/200 failures before, 0/200 after); `prune` guards against an empty valid set so a quarantine launch cannot wipe the OCR index
- **PerceptualHash** — uniform frames get a brightness-bucketed non-zero hash: identical solids still dedupe, visually distinct solids (black vs white slide) no longer collide, and the `0` "always keep" sentinel is never returned for a decodable image; non-uniform hashes stay bit-identical to main
- **TextGrabGeometry** — Vision boxes overflowing the unit square are trimmed to their visible portion rather than shifted onto unrelated text
- **ThumbnailGenerator** — no upscaling of small sources, zero-dimension guard, and ≥1 px clamping so extreme aspect ratios (e.g. 8000×30) no longer fail thumbnail generation on every timeline scroll
- **OverlayTimeline** — `formatRelativeTime` clamps future timestamps and all its day-boundary branches honour the injected `now` (deterministic tests; production behaviour unchanged)
- **FrameBuffer / TextCache / OverlayTimeline** — dependency injection (storage directory, `now:`) for testability

Also adds a shared Xcode scheme and a macOS unit-test CI workflow (triggers on `main` pushes and PRs).

## Review process

An initial multi-agent adversarial review of the branch confirmed 8 issues in the original sweep commits — including the quarantine path deferring rather than preventing data loss, and two new tests enshrining wrong behaviour as intended. Each finding was then independently re-verified by a second agent pass and fixed; the enshrining tests now assert the corrected behaviour.

## Test plan

- Full suite passes locally: **189 tests, 0 failures** (`xcodebuild test -scheme JustNow -destination 'platform=macOS'` with signing disabled, matching CI)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by yjsoon** [comment]
@coderabbitai review

**Comment by yjsoon** [comment]
All eight findings from the adversarial review were independently re-verified (each confirmed real) and are now fixed on this branch — see the seven new commits. Highlights:

- **Corrupt-manifest quarantine no longer defers data loss**: the frames directory is moved aside as `frames.corrupt` with the manifest, unreadable manifests take the same recovery path as undecodable ones, the quarantine is logged to the diagnostics log, and `TextCache.prune` can no longer wipe the OCR index on a quarantine launch.
- **Perceptual hash**: reverted to `>` with a brightness-bucketed non-zero fallback for uniform frames — identical solids still dedupe, black vs white no longer collide, and the `0` sentinel is never returned.
- Vision boxes are trimmed (not shifted) into the unit square; thumbnails clamp to ≥1 px per dimension; `formatRelativeTime` day branches honour the injected `now`; CI now triggers on `main` pushes and PRs; the whitespace-only normalisation case is actually tested.

Full suite: **189 tests, 0 failures** locally.

@coderabbitai review

File: JustNow/Storage/FrameStore.swift [hunk 1]
```diff
@@ -60,7 +65,19 @@ actor FrameStore {
...
+            } else {
+                // A corrupted manifest must not brick capture on every
+                // subsequent launch. Quarantine it for diagnosis and start
+                // fresh; `cleanupOrphans` deliberately refuses to delete
+                // frame files while the manifest is empty, so the on-disk
+                // JPEGs survive for manual recovery.
+                let quarantineURL = storageURL.appendingPathComponent("manifest.json.corrupt")
+                try? fileManager.removeItem(at: quarantineURL)
+                try? fileManager.moveItem(at: manifestURL, to: quarantineURL)
+                manifest = FrameManifest()
```
[the remark this answers is not in the document]
Comment (yjsoon) [reply JustNow/Storage/FrameStore.swift:81]: Fixed in 32b75d1: the quarantine now moves the frames directory aside as `frames.corrupt` (out of reach of orphan cleanup permanently, not just while the manifest is empty), `TextCache.prune` guards against an empty valid set so the quarantine launch no longer wipes the OCR index, and unreadable manifests take the same recovery path as undecodable ones. Covered by `testQuarantinedFramesSurviveCleanupAfterNewCapturesAreStored` and `testCorruptManifestDoesNotWipeTextCacheOnInit`.

File: JustNow/Capture/PerceptualHash.swift [hunk 2]
```diff
@@ -46,10 +46,15 @@ nonisolated struct PerceptualHash {
...
+        // Build hash: 1 if pixel >= mean, else 0. Using >= (not >) guarantees a
+        // non-zero hash for every decodable image (the maximum pixel is always
+        // >= the mean), which keeps 0 reserved as the "no hash" legacy sentinel
+        // that dedupe and timeline filtering treat as always-keep. A uniform
+        // frame hashing to 0 would otherwise bypass duplicate detection
+        // entirely and store every capture of a static solid-colour screen.
         var hash: UInt64 = 0
         for (index, pixel) in pixelData.enumerated() {
-            if pixel > mean {
+            if pixel >= mean {
```
[the remark this answers is not in the document]
Comment (yjsoon) [reply JustNow/Capture/PerceptualHash.swift]: Fixed in 6e96987: reverted to `>` (non-uniform hashes stay bit-identical to before) and only when the hash comes out 0 is a brightness-bucketed non-zero fallback synthesised — black hashes 0x1, white 0xFFFF (hamming distance 15), so distinct solids clear every threshold in use while identical solids still dedupe and the 0 sentinel is never returned. Covered by `testDistinctBrightnessUniformFramesDoNotDedupeAgainstEachOther`.

File: JustNow/Storage/FrameStore.swift [hunk 3]
```diff
@@ -42,31 +42,75 @@ actor FrameStore {
...
+    /// siblings, keeping the JPEGs out of reach of orphan cleanup.
+    private static func quarantineCorruptStore(
+        fileManager: FileManager,
+        storageURL: URL,
+        manifestURL: URL,
+        framesURL: URL,
+        reason: String
+    ) {
+        let manifestQuarantineURL = storageURL.appendingPathComponent("manifest.json.corrupt")
+        let framesQuarantineURL = storageURL.appendingPathComponent("frames.corrupt", isDirectory: true)
```
[the remark this answers is not in the document]
Comment (yjsoon) [reply JustNow/Storage/FrameStore.swift]: Good catch — fixed in 2ca65bd: `FrameStore.clear()` now also removes `manifest.json.corrupt` and `frames.corrupt`, so Clear All History deletes quarantined screenshots too. Covered by `testClearRemovesQuarantinedStore`.
